src/fr/bigsolo/src/eu/kanade/tachiyomi/extension/fr/bigsolo/BigSolo.kt

```
@@ -0,0 +1,266 @@
+
+        return if (groups != null) {
+            val keys = groups.keys()
+            if (keys.hasNext()) {
+                val firstGroupKey = keys.next()
+                val firstGroup = groups.optString(firstGroupKey)
+                firstGroup.substringAfterLast("/")
+            } else {
+                ""
+            }
+        } else {
+            ""
+        }
+    }
+
+    private fun buildChapterList(seriesData: SeriesData): List<SChapter> {
+        val chapters = seriesData.chapters ?: return emptyList()
+        val chapterList = mutableListOf<SChapter>()
+        val multipleChapters = chapters.size > 1
+
+        for ((chapterNumber, chapterData) in chapters) {
+            if (chapterData.licencied) continue
+
+            val title = chapterData.title ?: ""
+            val volumeNumber = chapterData.volume ?: ""
+
+            val baseName = if (multipleChapters) {
+                buildString {
+                    if (volumeNumber.isNotBlank()) append("Vol. $volumeNumber ")
+                    append("Ch. $chapterNumber")
+                    if (title.isNotBlank()) append(" – $title")
+                }
+            } else {
+                if (title.isNotBlank()) "One Shot – $title" else "One Shot"
+            }
+
+            val chapter = SChapter.create().apply {
+                name = baseName
+                url = "/${toSlug(seriesData.title)}/$chapterNumber"
+                chapter_number = chapterNumber.toFloatOrNull() ?: -1f
+                date_upload = (chapterData.lastUpdated ?: 0) * 1000L
+            }
+            chapterList.add(chapter)
+        }
+
+        return chapterList.sortedByDescending { it.chapter_number }
+    }
+
+    private fun fetchChapterPages(chapterId: String): List<Page> {
+        val pagesResponse = try {
+            client.newCall(GET("$baseUrl$CHAPTER_PAGES_API?id=$chapterId", headers)).execute()
+        } catch (e: java.io.IOException) {
```

dont swallow exceptions